src/types/ScreenInfoUiaProviderBase.cpp

```diff
@@ -96,31 +94,24 @@ ScreenInfoUiaProviderBase::Release()
 IFACEMETHODIMP ScreenInfoUiaProviderBase::QueryInterface(_In_ REFIID riid,
                                                          _COM_Outptr_result_maybenull_ void** ppInterface)
 {
+    RETURN_HR_IF_NULL(E_INVALIDARG, ppInterface);
+
     // TODO GitHub #1914: Re-attach Tracing to UIA Tree
     //Tracing::s_TraceUia(this, ApiCall::QueryInterface, nullptr);
-    if (riid == __uuidof(IUnknown))
-    {
-        *ppInterface = static_cast<IRawElementProviderSimple*>(this);
-    }
-    else if (riid == __uuidof(IRawElementProviderSimple))
+    if (riid == __uuidof(IUnknown) ||
+        riid == __uuidof(IRawElementProviderSimple) ||
+        riid == __uuidof(IRawElementProviderFragment) ||
+        riid == __uuidof(ITextProvider))
     {
-        *ppInterface = static_cast<IRawElementProviderSimple*>(this);
-    }
-    else if (riid == __uuidof(IRawElementProviderFragment))
-    {
-        *ppInterface = static_cast<IRawElementProviderFragment*>(this);
-    }
-    else if (riid == __uuidof(ITextProvider))
-    {
-        *ppInterface = static_cast<ITextProvider*>(this);
+        *ppInterface = this;

```

Comment:
this still scares me, but if it's okay, then .. i guess it's okay. only because all four of these interfaces _also_ derive from IUnknown. If they didn't, i think it would explode.